relayer-engine/src/storage/redisStore.ts

```diff
@@ -28,7 +28,8 @@ export class DefaultRedisWrapper implements RedisWrapper {
   }
 
   withRedis<T>(op: Op<T>): Promise<T> {
-    return this.redis.executeIsolated(op);
+    return op(this.redis)
```

Comment:
can't use executeIsolated and call one of the storage getters from another storage method or else we get redis deadlock

Reply:
I think we need `executeIsolated` for watch to work properly. Are you sure this doesn't break the isolation guarantees?